chore: export factory for builtin plugin

IWANABETHATGUY opened the pull request:
<!-- Thank you for contributing! -->

### Description

<!-- Please insert your description here and provide especially info about the "what" this PR is solving -->

hyf0 commented on packages/rolldown/tests/fixtures/builtin-plugin/glob-import/basic/_config.ts:
```diff
@@ -1,11 +1,11 @@
   config: {
     plugins: [
-      new BuiltinGlobImportPlugin({
+      builtinGlobImportPlugin({
```
I don't think we need to emphasize the `builtin` prefix here. These plugins are __imported__ from `rolldown`, which already has the builtin semantic.

Could you remove it? Other changes are fine.

IWANABETHATGUY committed:
chore: export factory for builtin plugin (#1808)

<!-- Thank you for contributing! -->

### Description

<!-- Please insert your description here and provide especially info about the "what" this PR is solving -->
